staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -365,6 +366,7 @@ func (b *Builder) URL(httpAttemptCount int, urls ...*url.URL) *Builder {
 // will be ignored (but logged at V(2)).
 func (b *Builder) Stdin() *Builder {
 	b.stream = true
+	b.stdin = true
```

Comment:
Note! There are other places where `kubectl` can potentially use standard input. I have not added code to protect against stdin fighting in these places because `kubectl` uses standard input once and then never looks at it again. Thus, `kubectl` would never try to use standard input when the exec plugin might be trying to use it.
- https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L283-L290
- https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/kubectl/pkg/rawhttp/raw.go#L58
- https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/kubectl/pkg/cmd/set/env/env_parse.go#L68
- https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/kubectl/pkg/cmd/replace/replace.go#L310
- Are there more????

How do we protect against someone adding more `kubectl` code that reads from stdin and may cause a stdin fight between `kubectl` and an exec plugin? I have no idea how to future-proof this???

Reply:
I do not know of a good way to keep this working perfectly "forever."

Reply:
- https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/kubectl/pkg/cmd/set/env/env_parse.go#L68
- https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/kubectl/pkg/cmd/replace/replace.go#L310


How can we hook these two uses up to the StdinUnavailable bits?

Reply:
`env_parse.go`: we could return a boolean from `envutil.ParseEnv()` for whether stdin was used in, and then wire that into the resource builder here (we would need a new `resource.Builder` function, which seems kinda leaky): https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go#L273

`replace.go`: if we did add a new `resource.Builder` function, we could call it again here; again, this seems a little leaky to me

Reply:
i added a number of commits to try to address @liggitt’s comments:
* 1a2957cebab: updated `resource.Builder` to contain function to say when stdin is in use (seems leaky, but best i could do)
* b5c4c593664: marked stdin as in use in `kubectl set env` (note - i ran into https://github.com/kubernetes/kubernetes/issues/102741 when trying to test this)
* f8599f88f9e: added cmd test for when someone tries to use stdin for both the resource and env vars in `kubectl set env`; because of these new `resource.Builder` changes, users will see a new error on the CLI that will prevent them from shooting themselves in the foot:
```akeesler@akeesler-a02:kubernetes$ echo a=b | ./_output/local/bin/darwin/amd64/kubectl set env -f - -
error: standard input cannot be used for multiple arguments```
* e5af8c48475: marked stdin as in use in `kubectl replace --force`
* 441ab1784f3: added cmd tests for when these commands are used with `Always` interactive exec plugin

Reply:
those changes look great, thanks. one nit on https://github.com/kubernetes/kubernetes/commit/b5c4c5936642e6f05ded12d60cfb2f9775cba8b2#r52149752 to track/return whether stdin was consumed even in error cases

Reply:
nit addressed in https://github.com/kubernetes/kubernetes/pull/99310/commits/20193d72149e81a410334e3c4145ab57b9b40687